client/app/components/routes/select-operator/form/UserOperator.tsx

```diff
@@ -53,38 +58,47 @@ const createUserOperatorSchema = (
         .pc_business_structure,
       anyOf: businessStructureOptions,
     };
+
   return localSchema;
 };
 
 export default async function UserOperator({
   params,
 }: Readonly<{
-  params?: { id?: number };
+  params?: { id?: number; readonly?: boolean };
 }>) {
-  const serverError = <div>Server Error. Please try again later.</div>;
-
   const businessStructures: BusinessStructure[] | { error: string } =
     await getBusinessStructures();
 
   const userData: UserInformationInitialFormData | { error: string } =
     await getCurrentUser();
 
-  if ("error" in userData || "error" in businessStructures) return serverError;
+  // TODO: define schema of data returned from endpoint
+  const userOperatorData: any | { error: string } =
+    await getUserOperatorFormData(params.id);
+
+  if (
+    "error" in userData ||
+    "error" in businessStructures ||
+    "error" in userOperatorData
+  )
+    return;
 
   const businessStructuresList = businessStructures?.map(
     (businessStructure: BusinessStructure) => ({
       id: businessStructure.name,
       label: businessStructure.name,
     }),
   );
-
+  // FIXME: this data is bogus. Replace with genuine data.
+  userOperatorData.is_senior_officer = "true";
+  userOperatorData.operator_has_parent_company = "no";
```

Comment:
these 2 lines need to be fixed

Reply:
These lines have been removed I believe

Reply:
@Sepehr-Sobhani I think this is the main problematic TODO, we need to store these values or use a resolver as you previously mentioned.

Reply:
Actually I think this is good to be removed, the form saves fine without it.